controllers/secret/secret_controller.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package secret
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	"github.com/vmware-samples/load-balancer-operator-for-kubernetes/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"os"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SetupWithManager adds this reconciler to a new controller then to the
+// provided manager.
+func (r *SecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		// Watch ConfigMap resources.
+		For(&corev1.Secret{}).
+		Complete(r)
+}
+
+// SecretReconciler reconciles avi-credentials and avi-ca secrets
+type SecretReconciler struct {
+	client.Client
+	Log              logr.Logger
+	Scheme           *runtime.Scheme
+	CredentialInited bool //avi-controller-credential secret read once after bootup
+	CAInited         bool //avi-controller-ca secret read once after bootup
+}
+
+// AKOO will reconcile default secrets avi-controller-credentials and avi-controller-ca
+// If customer uses their own secrets, AKOO has to be restarted when secrets are updated
+func (r *SecretReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
Instead of shutting down the `ako-operator`, I believe the better solution is:
when we detect the new secret values, we:
1. re-init avi client in `ako-operator`
2. re-generate add-on secret of each ako deployment

Those two can be implemented by re-trigger the `akodeploymentconfig_controller` to reconcile, but you may need to change the logic on how to init avi client in `akodeploymentconfig_controller` now, cause it is lazy init now.
Re-trigger can be easily done via a mapper function like this: 
https://github.com/vmware-samples/load-balancer-operator-for-kubernetes/blob/main/controllers/cluster/cluster_controller.go#L38
https://github.com/vmware-samples/load-balancer-operator-for-kubernetes/blob/main/pkg/controller-runtime/handlers/service_for_cluster_handler.go#L29

Reply:
I thought of restarting ako-operator since I thought it might be a bigger effort to reinitialize avi-client. Also noticed that AKO uses similar approach of restart. I dont have enough supporting info if restarting is a common approach in stateless apps like ako-operator

I agree its not elegant to shutdown. I can investigate the efforts to avoid that

Reply:
Done. Please review the latest